starcheck/utils.py

```diff
@@ -17,119 +29,113 @@
 from starcheck.pcad_att_check import make_pcad_attitude_check_report
 from starcheck.plot import make_plots_for_obsid
 
+ACQS = mica.stats.acq_stats.get_stats()
+GUIDES = mica.stats.guide_stats.get_stats()
 
-def python_from_perl(func):
-    """Decorator to facilitate calling Python from Perl inline.
 
-    - Convert byte strings to unicode.
-    - Print stack trace on exceptions which perl inline suppresses.
-    """
-    @functools.wraps(func)
-    def wrapper(*args, **kwargs):
-        args = de_bytestr(args)
-        kwargs = de_bytestr(kwargs)
-        try:
-            return func(*args, **kwargs)
-        except Exception:
-            import traceback
-            traceback.print_exc()
-            raise
-    return wrapper
+def date2secs(val):
+    """Convert date to seconds since 1998.0"""
+    out = cxotime.date2secs(val)
```

Comment:
I note in passing that Chandra.Time still seems to be winning on speed a bit for these very limited cases.  If we are planning to modernize it as a non-namespace package, is there any value in keeping it around in "new" code?

Reply:
It's important to profile and quantify the effect on overall program execution.

The Chandra.Time version is only about 5 µs faster (16 vs 21). Which means you need about a million calls to that function in order to have an appreciable impact on a runtime that is around 60 seconds. So basically the answer is "No, we should never use it in new code unless you can demonstrate through profiling that it matters".

(One of my performance PR's does currently use DateTime, but I'll be fixing that with the `convert_time_format` function which is fast enough.)

Reply:
I did this profiling and my conclusion was just "if we are making a change, why are we picking the slower one".  If your point is that cxotime is the supported code moving forward that's fine... I just saw this working going into chandra_time as well and was trying to figure out the reason for picking the slower code.

By my profiling I agree that there's no overall effect on program execution.

Reply:
I picked `Chandra.Time` as a stressing test case for de-namespace because it has a compiled C++ code, not so much because I want to modernize / endorse it. But eventually *every* namespace package will get the treatment, I just didn't want to discover a problem late in the process.